### PR TITLE
DPDK: sriov rescind test fixes

### PR DESCRIPTION
**Pull request by mcgov**
After comparing rescind test failures across kernel+dpdk+rdma_core+host verisons,
apply some fixes based on the investigation.

- Switch to DPDK 21.11 on Ubuntu 18.04 (more stable across test matrix)
- Remove dmesg wait and async timeout
- Add secondary regex for hotplug to cover EAL version differences in
output
- Add forceful quit for unresponsive (but functional) testpmd by reloading
hv_netvsc

### CODE REVIEW

**Comment by squirrelsc** [comment]
@LiliDeng please review and merge.

**Comment by mcgov** [comment]
Tested on Ubuntu 18.04 & 20.04, Centos 7.5 & 7.7, Rhel 8.0 & 8.3 on Standard_DS4_v2. Stress tested Ubuntu 18.04 to verify it works across host versions (20x iterations).
Running additional tests on different SKUs now.

edit: . Rescind on v2 and v3 both look good, hitting an issue with v4 and v5 where SKUs are reporting GPU and Infiniband feature capability which blocks the test.

**Comment by mcgov** [comment]
adding Kameron for visibility, will wait on review from Lili to merge

**Comment by kamalca** [comment]
@mcgov This updates Ubuntu 18.04 DPDK version, potentially fixing other tests with versioning issues on 18.04, correct? Have you tested the effects of the version change on other tests?

**Comment by mcgov** [comment]
> @mcgov This updates Ubuntu 18.04 DPDK version, potentially fixing other tests with versioning issues on 18.04, correct? Have you tested the effects of the version change on other tests?

Good thought! I'll run the full suite across SKUs

File: microsoft/testsuites/dpdk/dpdkutil.py
Comment (LiliDeng) [inline microsoft/testsuites/dpdk/dpdkutil.py:287]: above switch_sriov already use wait, so it will check the rescind works or not there, no need sleep

File: microsoft/testsuites/dpdk/dpdkutil.py
Comment (LiliDeng) [inline microsoft/testsuites/dpdk/dpdkutil.py:294]: same

File: microsoft/testsuites/dpdk/dpdkutil.py
Comment (LiliDeng) [inline microsoft/testsuites/dpdk/dpdkutil.py]: above two patterns are not used?

File: microsoft/testsuites/dpdk/dpdkutil.py
Comment (mcgov) [reply microsoft/testsuites/dpdk/dpdkutil.py:287]: I do need to collect some data with each state though, so these sleeps are to let the other thread run for a few seconds.

File: microsoft/testsuites/dpdk/dpdkutil.py
Comment (mcgov) [reply microsoft/testsuites/dpdk/dpdkutil.py]: Edit: removed code that used these but left the pattern! Will remove

Comment (mcgov) [reply microsoft/testsuites/dpdk/dpdkutil.py]: removed :)

File: microsoft/testsuites/dpdk/dpdkutil.py
Comment (mcgov) [reply microsoft/testsuites/dpdk/dpdkutil.py:294]: same as above, sleeps are to let testpmd run for a few seconds on the other thread to gather data.

**Comment by LiliDeng** [comment]
@mcgov in initialize_node_resources, I saw there is a check in network interface enabling sriov or not, we also need check in guest side, like whether every nic has VF nic paired, since I saw failures caused by VF not initialized in the guest and enable_uio_hv_generic_for_nic was invoked, then case failed for below reason.
```
AssertionError: [This interface eth1 does not have a lower interface and pci slot associated with it. Aborting.] Expected <> to be not equal to <>, but was
```

**Comment by mcgov** [comment]
> @mcgov in initialize_node_resources, I saw there is a check in network interface enabling sriov or not, we also need check in guest side, like whether every nic has VF nic paired, since I saw failures caused by VF not initialized in the guest and enable_uio_hv_generic_for_nic was invoked, then case failed for below reason.
> 
> ```
> AssertionError: [This interface eth1 does not have a lower interface and pci slot associated with it. Aborting.] Expected <> to be not equal to <>, but was
> ```

Sure I'll add it 😄

**Comment by mcgov** [comment]
Additional changes are incoming, hold merging please

**Comment by mcgov** [comment]
Updated, ready for review from @LiliDeng

**Comment by mcgov** [comment]
@LiliDeng feel free to resolve conversations and merge if it looks good.

**Comment by LiliDeng** [comment]
Test results
```
2022-07-19 03:34:03.299[28344][INFO] lisa.RootRunner                      Dpdk.verify_dpdk_build_netvsc: PASSED
2022-07-19 03:34:03.300[28344][INFO] lisa.RootRunner                    Dpdk.verify_dpdk_build_failsafe: PASSED
2022-07-19 03:34:03.301[28344][INFO] lisa.RootRunner                               Dpdk.verify_dpdk_ovs: FAILED   failed. AttributeError: 'DpdkTestpmd' object has no attribute '_dpdk_version_info'
2022-07-19 03:34:03.301[28344][INFO] lisa.RootRunner                            Dpdk.verify_dpdk_nff_go: PASSED
2022-07-19 03:34:03.302[28344][INFO] lisa.RootRunner                      Dpdk.verify_dpdk_multiprocess: SKIPPED  skipped: Unsupported Hardware: ConnectX3 does not support secondary process RX
2022-07-19 03:34:03.303[28344][INFO] lisa.RootRunner   Dpdk.verify_dpdk_sriov_rescind_failover_receiver: PASSED
2022-07-19 03:34:03.304[28344][INFO] lisa.RootRunner  Dpdk.verify_dpdk_sriov_rescind_failover_send_only: PASSED
2022-07-19 03:34:03.305[28344][INFO] lisa.RootRunner                               Dpdk.verify_dpdk_vpp: PASSED
2022-07-19 03:34:03.305[28344][INFO] lisa.RootRunner                         Dpdk.verify_dpdk_ring_ping: FAILED   failed. AssertionError: [RingPing measured 4175900 as maximum ping latency, maximum should be less than 200000] Expected <4175900> to be less than <200000>, but was not.
2022-07-19 03:34:03.306[28344][INFO] lisa.RootRunner Dpdk.verify_dpdk_send_receive_multi_txrx_queue_failsafe: PASSED
2022-07-19 03:34:03.307[28344][INFO] lisa.RootRunner Dpdk.verify_dpdk_send_receive_multi_txrx_queue_netvsc: PASSED
2022-07-19 03:34:03.308[28344][INFO] lisa.RootRunner             Dpdk.verify_dpdk_send_receive_failsafe: PASSED
2022-07-19 03:34:03.308[28344][INFO] lisa.RootRunner               Dpdk.verify_dpdk_send_receive_netvsc: PASSED
2022-07-19 03:34:03.309[28344][INFO] lisa.RootRunner                            Dpdk.verify_uio_binding: PASSED
2022-07-19 03:34:03.310[28344][INFO] lisa.RootRunner test result summary
2022-07-19 03:34:03.311[28344][INFO] lisa.RootRunner     TOTAL    : 14
2022-07-19 03:34:03.312[28344][INFO] lisa.RootRunner     QUEUED   : 0
2022-07-19 03:34:03.313[28344][INFO] lisa.RootRunner     ASSIGNED : 0
2022-07-19 03:34:03.313[28344][INFO] lisa.RootRunner     RUNNING  : 0
2022-07-19 03:34:03.314[28344][INFO] lisa.RootRunner     FAILED   : 2
2022-07-19 03:34:03.315[28344][INFO] lisa.RootRunner     PASSED   : 11
2022-07-19 03:34:03.316[28344][INFO] lisa.RootRunner     SKIPPED  : 1
2022-07-19 03:34:03.321[28344][INFO] lisa. completed in 2080.393 sec
```

**Review by LiliDeng — APPROVED** [review]

**Pull request merged**
